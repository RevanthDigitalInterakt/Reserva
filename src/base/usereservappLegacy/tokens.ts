export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
