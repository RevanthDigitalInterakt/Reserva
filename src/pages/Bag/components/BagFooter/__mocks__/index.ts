import { profileQuery } from '../../../../../graphql/profile/profileQuery';

export const bagInfos = {
  totalBagItemsPrice: 100,
  totalBagItems: 20,
  totalBagDiscountPrice: 40,
  totalBagDeliveryPrice: 60,
};

export const currentOrderForm: any = {
  items: [
    {
      name: 'Produto A',
      skuName: 'test',
      price: 50.0,
      quantity: 2,
      productId: 4,
      productCategories: [
        {},
      ],
      additionalInfo: {
        brandName: 'RESERVA',
      },
    },
    {
      name: 'Produto B',
      skuName: 'test',
      price: 20.0,
      quantity: 1,
      productId: 4,
      productCategories: [
        {},
      ],
      additionalInfo: {
        brandName: 'RESERVA',
      },
    },
  ],
  orderFormId: '1',
};

export const installmentInfo = {
  installmentPrice: 110,
  installmentsNumber: 40,
  totalPrice: 1000,
};

export const apolloMocks = [
  {
    request: {
      query: profileQuery,
      variables: {},
    },
    result: {
      data: {
        profile: {
          __typename:
            'vtex_storegraphql_2_160_0_Profile',
          addresses: [[{}]],
          customFields: [[{}], [{}], [{}]],
          userId: '4d35c3cf-d4a1-4c52-a421-66d8f97f1b10',
          firstName: 'Tester',
          lastName: 'Silva',
          fullName: 'teste da Silva',
          email: 'nogueirahy@gmail.com',
          document: '11111111111',
          birthDate: '1980-01-01T00:00:00.000Z',
          gender: 'male',
          homePhone: '+5511991111111',
          payments: null,
          passwordLastUpdate: 'teste123',
        },
      },
    },
  },
];

export const apolloMocksWithoutDataUser = [
  {
    request: {
      query: profileQuery,
      variables: {},
    },
    result: {
      data: {
        profile: {
          __typename:
            'vtex_storegraphql_2_160_0_Profile',
          userId: '4d35c3cf-d4a1-4c52-a421-66d8f97f1b10',
          firstName: null,
          lastName: null,
          fullName: null,
          email: 'nogueirahy@gmail.com',
          document: null,
          birthDate: null,
          gender: 'male',
          homePhone: null,
          payments: null,
          passwordLastUpdate: 'teste123',
        },
      },
    },
  },
];
