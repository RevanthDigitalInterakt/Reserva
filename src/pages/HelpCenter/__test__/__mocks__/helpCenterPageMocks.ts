import type { DocumentNode } from 'graphql';
import { HelpCenterCollectionDocument, type HelpCenterCollectionQuery } from '../../../../base/graphql/generated';
import { mockHelpCenterCollectionData } from '../../../../zustand/useHelpCenterStore/mocks/helpCenterCollection.mock';

interface IApolloMock<T> {
  request: {
    query: DocumentNode,
    variables: object,
  },
  result: {
    data: T
  }
}

export const helpCenterPageMocks: Array<IApolloMock<HelpCenterCollectionQuery>> = [
  {
    request: {
      query: HelpCenterCollectionDocument,
      variables: {},
    },
    result: {
      data: {
        helpCenterCollection: {
          ...mockHelpCenterCollectionData,
        },
        __typename: 'Query',
      },
    },
  },
];
