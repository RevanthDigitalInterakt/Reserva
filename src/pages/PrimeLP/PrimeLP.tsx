import type { StackScreenProps } from '@react-navigation/stack';
import React, {
  useCallback, useMemo, useState,
} from 'react';
import { ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { useLandingPagePrimeQuery } from '../../base/graphql/generated';
import { ModalBag } from '../../components/ModalBag/ModalBag';
import { useApolloFetchPolicyStore } from '../../zustand/useApolloFetchPolicyStore';
import testProps from '../../utils/testProps';
import { ModalWelcomePrime } from '../../components/ModalWelcomePrime';
import { usePrimeInfo } from '../../hooks/usePrimeInfo';
import { TopBarDefaultBackButton } from '../../modules/Menu/components/TopBarDefaultBackButton';
import type { RootStackParamList } from '../../routes/StackNavigator';
import { usePrimeStore } from '../../zustand/usePrimeStore/usePrimeStore';
import PrimeBenefits from './components/PrimeBenefits';
import PrimeFAQ from './components/PrimeFAQ/PrimeFAQ';
import PrimeHero from './components/PrimeHero';
import PrimeIntro from './components/PrimeIntro';
import PrimeSubscribe from './components/PrimeSubscribe';
import { Box } from '../../components/Box/Box';
import { ExceptionProvider } from '../../base/providers/ExceptionProvider';

type IPrimeLP = StackScreenProps<RootStackParamList, 'PrimeLP'>;

function PrimeLP({ navigation }: IPrimeLP) {
  const { onAddPrimeToCart, isPrime } = usePrimeInfo();
  const { getFetchPolicyPerKey } = useApolloFetchPolicyStore(['getFetchPolicyPerKey']);
  const [loadingAddCartPrime, setLoadingAddCartPrime] = useState(false);

  const {
    animationBag,
    handleClickContinue,
    isVisibleModalWelcome,
    changeStateAnimationBag,
    changeStateIsVisibleModalWelcome,
  } = usePrimeStore([
    'animationBag',
    'handleClickContinue',
    'isVisibleModalWelcome',
    'changeStateAnimationBag',
    'changeStateIsVisibleModalWelcome',
  ]);

  const { data: rawData, loading } = useLandingPagePrimeQuery({
    context: { clientName: 'gateway' },
    fetchPolicy: getFetchPolicyPerKey('landingPagePrime'),
  });

  const data = useMemo(() => rawData?.landingPagePrime, [rawData?.landingPagePrime]);

  const handleOnModalHideSignIn = useCallback(() => {
    changeStateAnimationBag(true);
  }, [changeStateAnimationBag]);

  const handleOnModalHide = useCallback(() => {
    if (isPrime) {
      setTimeout(() => {
        changeStateIsVisibleModalWelcome(true);
      }, 500);
    }
  }, [changeStateIsVisibleModalWelcome, isPrime]);

  const onAddToCart = useCallback(async () => {
    try {
      if (!data || loadingAddCartPrime) return;

      if (isPrime) {
        navigation.popToTop();
        return;
      }

      setLoadingAddCartPrime(true);

      await onAddPrimeToCart(false);

      handleOnModalHideSignIn();
    } catch (e) {
      ExceptionProvider.captureException(e);
    } finally {
      setLoadingAddCartPrime(false);
    }
  }, [data, handleOnModalHideSignIn, isPrime, loadingAddCartPrime, navigation, onAddPrimeToCart]);

  const onCloseModalWelcomePrime = useCallback(() => {
    handleClickContinue();

    setTimeout(() => {
      navigation.navigate('Home');
    }, 300);
  }, [handleClickContinue, navigation]);

  return (
    <SafeAreaView style={{ backgroundColor: '#fff' }}>
      <Box bg="white">
        <TopBarDefaultBackButton
          loading={loading || loadingAddCartPrime}
          navigateGoBack
        />

        <ModalBag
          isVisible={animationBag}
          onBackdropPress={() => changeStateAnimationBag(false)}
          onModalHide={handleOnModalHide}
        />

        <ModalWelcomePrime
          isVisible={isVisibleModalWelcome}
          onClose={onCloseModalWelcomePrime}
        />

        {!!(data && !loading) && (
          <ScrollView
            contentContainerStyle={{ paddingBottom: 100 }}
            {...testProps('PrimeLP_page')}
          >
            <PrimeHero data={data} onAddToCart={onAddToCart} />

            <PrimeIntro data={data} />

            <PrimeBenefits data={data} />

            <PrimeSubscribe data={data} onAddToCart={onAddToCart} />

            <PrimeFAQ data={data} />
          </ScrollView>
        )}
      </Box>
    </SafeAreaView>
  );
}

export default PrimeLP;
