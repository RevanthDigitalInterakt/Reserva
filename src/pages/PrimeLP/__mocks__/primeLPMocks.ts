import type { DocumentNode } from 'graphql';
import { LandingPagePrimeDocument, LandingPagePrimeQuery } from '../../../base/graphql/generated';
import { mockPrimeData } from '../../../../__mocks__/PrimeLP.mock';

interface IApolloMock<T> {
  request: {
    query: DocumentNode,
    variables: object,
  },
  result: {
    data: T
  }
}

export const ApolloMockLPPrime: Array<IApolloMock<LandingPagePrimeQuery>> = [
  {
    request: {
      query: LandingPagePrimeDocument,
      variables: {},
    },
    result: {
      data: {
        landingPagePrime: {
          ...mockPrimeData,
        },
        __typename: 'Query',
      },
    },
  },
];
