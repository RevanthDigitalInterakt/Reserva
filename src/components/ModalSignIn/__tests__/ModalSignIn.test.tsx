/* eslint-disable react/jsx-no-constructed-context-values */
import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { render, screen } from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import { ModalSignIn } from '..';
import { ApolloMockLPPrime } from '../../../pages/PrimeLP/__mocks__/primeLPMocks';
import * as useLandingPagePrimeQuery from '../../../base/graphql/generated';
import { mockPrimeData } from '../../../../__mocks__/PrimeLP.mock';
import { theme } from '../../../base/usereservappLegacy/theme';

jest.mock('../../../zustand/useApolloFetchPolicyStore', () => ({
  useApolloFetchPolicyStore: () => ({
    getFetchPolicyPerKey: () => 'network-only',
  }),
}));

jest.mock('../../../zustand/useDitoStore', () => ({
  useDitoStore: () => ({
  }),
}));

jest
  .spyOn(useLandingPagePrimeQuery, 'useLandingPagePrimeQuery')
  .mockReturnValue({
    data: {
      landingPagePrime: {
        ...mockPrimeData,
      },
    },
    loading: false,
  } as any);

const closeModal = jest.fn();

function TestingComponent() {
  return (
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={ApolloMockLPPrime} addTypename={false}>
          <ModalSignIn isVisible onClose={closeModal} />
      </MockedProvider>
    </ThemeProvider>
  );
}

describe('ModalSignIn', () => {
  beforeEach(() => {
    jest.useFakeTimers({ legacyFakeTimers: true });
  });

  it('should render properly', () => {
    render(TestingComponent());

    const titleModal = screen.getByTestId('com.usereserva:id/modal_sign_in_title');

    expect(titleModal).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    render(TestingComponent());
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
