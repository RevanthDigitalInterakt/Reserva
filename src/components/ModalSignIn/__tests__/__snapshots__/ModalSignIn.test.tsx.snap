// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSignIn should match to snapshot 1`] = `
<Modal
  accessibilityLabel="com.usereserva:id/modal_sign_in"
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="com.usereserva:id/modal_sign_in"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    accessibilityLabel="com.usereserva:id/modal_sign_in"
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      accessibilityLabel="com.usereserva:id/modal_sign_in_container"
      backgroundColor="white"
      p="xxxs"
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          Object {
            "top": -60,
            "width": 702,
          },
        ]
      }
      testID="com.usereserva:id/modal_sign_in_container"
    >
      <View
        flexDirection="row-reverse"
        style={
          Array [
            Object {
              "flexDirection": "row-reverse",
            },
          ]
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/modal_sign_in_close"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="com.usereserva:id/modal_sign_in_close"
        >
          <RNSVGSvgView
            bbHeight={14}
            bbWidth={14}
            color="preto"
            focusable={false}
            size={14}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "color": "#333333",
                    "height": 14,
                    "width": 14,
                  },
                ],
                Object {
                  "flex": 0,
                  "height": 14,
                  "width": 14,
                },
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                    ],
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={8.24}
                vbWidth={8.242}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M5.098 4.12l2.944-2.944A.69.69 0 007.066.2L4.122 3.144 1.178.2a.69.69 0 10-.976.976L3.146 4.12.202 7.064a.69.69 0 00.976.976l2.944-2.944L7.066 8.04a.69.69 0 00.976-.976z"
                    fill={4281545523}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <Text
        accessibilityLabel="com.usereserva:id/modal_sign_in_title"
        fontFamily="reservaSerifBlack"
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 27,
            },
          ]
        }
        testID="com.usereserva:id/modal_sign_in_title"
        variant="descontoTag1"
      >
        Prime
      </Text>
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View
            style={
              Array [
                Object {},
                Object {
                  "marginBottom": 24,
                  "marginTop": 24,
                },
              ]
            }
          >
            <Text
              color="neutroFrio2"
              fontSize={14}
              style={
                Array [
                  Object {
                    "color": "#8A8C8E",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Olá! Já é cliente Prime? Então basta inserir seu e-mail abaixo para poder comprar:
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 710,
                  },
                ]
              }
              width={710}
            >
              <View
                borderBottomColor="neutroFrio2"
                borderBottomWidth="hairline"
                flexDirection="row"
                justifyContent="space-between"
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#8A8C8E",
                      "borderBottomWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                    Object {
                      "overflow": "hidden",
                    },
                  ]
                }
              >
                <View
                  flexGrow={4}
                  style={
                    Array [
                      Object {
                        "flexGrow": 4,
                      },
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="com.usereserva:id/modal_sign_in_input_email"
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    collapsable={false}
                    keyboardType="email-address"
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onGestureHandlerEvent={[Function]}
                    onGestureHandlerStateChange={[Function]}
                    placeholder="Digite seu e-mail"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "margin": 0,
                        "maxWidth": 710,
                        "padding": 0,
                      }
                    }
                    testID="com.usereserva:id/modal_sign_in_input_email"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              mt="md"
              style={
                Array [
                  Object {
                    "marginTop": 48,
                    "width": "100%",
                  },
                ]
              }
              width="100%"
            >
              <View
                style={
                  Array [
                    Object {
                      "width": 710,
                    },
                  ]
                }
                width={710}
              >
                <View
                  borderBottomColor="neutroFrio2"
                  borderBottomWidth="hairline"
                  flexDirection="row"
                  justifyContent="space-between"
                  style={
                    Array [
                      Object {
                        "borderBottomColor": "#8A8C8E",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                      Object {
                        "overflow": "hidden",
                      },
                    ]
                  }
                >
                  <View
                    flexGrow={4}
                    style={
                      Array [
                        Object {
                          "flexGrow": 4,
                        },
                      ]
                    }
                  >
                    <TextInput
                      accessibilityLabel="com.usereserva:id/modal_sign_in_input_password"
                      allowFontScaling={true}
                      autoCapitalize="none"
                      autoCompleteType="off"
                      autoCorrect={false}
                      collapsable={false}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                      placeholder="Digite sua senha"
                      rejectResponderTermination={true}
                      secureTextEntry={true}
                      style={
                        Object {
                          "margin": 0,
                          "maxWidth": 684,
                          "padding": 0,
                        }
                      }
                      testID="com.usereserva:id/modal_sign_in_input_password"
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                  <View
                    justifyContent="center"
                    mr="xxxs"
                    style={
                      Array [
                        Object {
                          "justifyContent": "center",
                          "marginRight": 16,
                        },
                        Object {
                          "width": 40,
                        },
                      ]
                    }
                  >
                    <View
                      alignItems="center"
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "opacity": 1,
                          }
                        }
                        testID="com.usereserva:id/modal_sign_in_input_password_button_hide_password"
                      >
                        <RNSVGSvgView
                          bbHeight={22}
                          bbWidth={22}
                          color="preto"
                          focusable={false}
                          size={22}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              Array [
                                Object {
                                  "color": "#333333",
                                  "height": 22,
                                  "width": 22,
                                },
                              ],
                              Object {
                                "flex": 0,
                                "height": 22,
                                "width": 22,
                              },
                            ]
                          }
                        >
                          <RNSVGGroup>
                            <RNSVGSvgView
                              align="xMidYMid"
                              bbHeight="100%"
                              bbWidth="100%"
                              fill="none"
                              focusable={false}
                              height="100%"
                              meetOrSlice={0}
                              minX={0}
                              minY={0}
                              stroke="#333333"
                              strokeLinecap="round"
                              strokeLinejoin="round"
                              strokeWidth={2}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "borderWidth": 0,
                                  },
                                  Array [
                                    Object {
                                      "height": "100%",
                                      "width": "100%",
                                    },
                                  ],
                                  Object {
                                    "flex": 0,
                                    "height": "100%",
                                    "width": "100%",
                                  },
                                ]
                              }
                              vbHeight={24}
                              vbWidth={24}
                              width="100%"
                            >
                              <RNSVGGroup
                                fill={null}
                                propList={
                                  Array [
                                    "fill",
                                    "stroke",
                                    "strokeWidth",
                                    "strokeLinecap",
                                    "strokeLinejoin",
                                  ]
                                }
                                stroke={4281545523}
                                strokeLinecap={1}
                                strokeLinejoin={1}
                                strokeWidth={2}
                              >
                                <RNSVGPath
                                  d="M17.94 17.94A10.07 10.07 0 0112 20c-7 0-11-8-11-8a18.45 18.45 0 015.06-5.94M9.9 4.24A9.12 9.12 0 0112 4c7 0 11 8 11 8a18.5 18.5 0 01-2.16 3.19m-6.72-1.07a3 3 0 11-4.24-4.24"
                                />
                                <RNSVGPath
                                  d="M1 1L23 23"
                                />
                              </RNSVGGroup>
                            </RNSVGSvgView>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                mb="quarck"
                mt="micro"
                style={
                  Array [
                    Object {
                      "marginBottom": 4,
                      "marginTop": 12,
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="com.usereserva:id/modal_sign_in_cta_forgot_password"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="com.usereserva:id/modal_sign_in_cta_forgot_password"
                >
                  <Text
                    fontSize={14}
                    style={
                      Array [
                        Object {
                          "fontSize": 14,
                        },
                        Object {
                          "textDecorationLine": "underline",
                        },
                      ]
                    }
                  >
                    Esqueci minha senha
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            alignItems="stretch"
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "color": "#FFF",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 13,
                  "justifyContent": "center",
                  "marginBottom": 24,
                  "marginTop": 32,
                  "minWidth": 130,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 16,
                }
              }
              testID="com.usereserva:id/modal_sign_in-cta_enter"
            >
              <View
                alignItems="center"
                flexDirection="row"
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    color="white"
                    fontFamily="nunitoBold"
                    fontSize={13}
                    letterSpacing={1.6}
                    style={
                      Array [
                        Object {
                          "color": "#FFF",
                          "fontFamily": "Nunito-Bold",
                          "fontSize": 13,
                          "letterSpacing": 1.6,
                        },
                      ]
                    }
                  >
                    CONTINUAR COMPRANDO
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            alignItems="center"
            flexDirection="row"
            justifyContent="center"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "marginBottom": 24,
                  "marginTop": 8,
                },
              ]
            }
          >
            <View
              borderColor="divider"
              borderWidth="hairline"
              flex={1}
              marginRight="md"
              style={
                Array [
                  Object {
                    "borderColor": "#D8D9DA",
                    "borderWidth": 1,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginRight": 48,
                  },
                ]
              }
            />
            <Text
              fontSize={14}
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                ]
              }
              textAlign="center"
            >
              Ainda não é
               
              <Text
                fontFamily="reservaSerifBlack"
                fontSize={14}
                style={
                  Array [
                    Object {
                      "fontFamily": "ReservaSerif-Black",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Prime
              </Text>
              ?
            </Text>
            <View
              borderColor="divider"
              borderWidth="hairline"
              flex={1}
              marginLeft="md"
              style={
                Array [
                  Object {
                    "borderColor": "#D8D9DA",
                    "borderWidth": 1,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginLeft": 48,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              color="neutroFrio2"
              fontFamily="reservaSansRegular"
              fontSize={14}
              style={
                Array [
                  Object {
                    "color": "#8A8C8E",
                    "fontFamily": "ReservaSans-Regular",
                    "fontSize": 14,
                  },
                  Object {
                    "marginBottom": 24,
                  },
                ]
              }
            >
              Por apenas
               
              <Text
                color="fullBlack"
                fontFamily="reservaSansBold"
                fontSize={14}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "ReservaSans-Bold",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                12
                x R$
                 
                25
              </Text>
              , assinantes
               
              <Text
                color="fullBlack"
                fontFamily="reservaSerifBlack"
                fontSize={14}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "ReservaSerif-Black",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Prime
              </Text>
               
              têm acesso a vantagens como descontos exclusivos e frete grátis em compras acima de
               
              <Text
                color="fullBlack"
                fontFamily="reservaSansBold"
                fontSize={14}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "ReservaSans-Bold",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                R$
                 
                499
              </Text>
              , toque no botão abaixo e comece agora mesmo a comprar!
            </Text>
            <View
              alignItems="stretch"
              style={
                Array [
                  Object {
                    "alignItems": "stretch",
                  },
                ]
              }
            >
              <View
                accessibilityLabel="com.usereserva:id/modal_sign_in_cta_add_prime"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#11AB6B",
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": "100%",
                  }
                }
                testID="com.usereserva:id/modal_sign_in_cta_add_prime"
              >
                <View
                  alignItems="center"
                  height={48}
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 48,
                        "justifyContent": "center",
                        "width": "100%",
                      },
                    ]
                  }
                  width="100%"
                >
                  <Text
                    color="white"
                    fontFamily="reservaSansMedium"
                    fontSize={14}
                    style={
                      Array [
                        Object {
                          "color": "#FFF",
                          "fontFamily": "ReservaSans-Medium",
                          "fontSize": 14,
                        },
                      ]
                    }
                  >
                    ADICIONAR PRIME AO CARRINHO
                  </Text>
                </View>
              </View>
            </View>
            <Text
              color="neutroFrio2"
              fontSize={12}
              style={
                Array [
                  Object {
                    "color": "#8A8C8E",
                    "fontSize": 12,
                  },
                  Object {
                    "marginTop": 12,
                  },
                ]
              }
            >
              Para mais informações sobre o prime,
               
              <Text
                accessibilityLabel="com.usereserva:id/modal_sign_click_here"
                color="fullBlack"
                fontFamily="reservaSansBold"
                fontSize={12}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "ReservaSans-Bold",
                      "fontSize": 12,
                    },
                    Object {
                      "textDecorationLine": "underline",
                    },
                  ]
                }
                testID="com.usereserva:id/modal_sign_click_here"
              >
                clique aqui
              </Text>
               
              antes de concluir seu pedido.
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
